modules/rke2/variables.tf

```
@@ -0,0 +1,127 @@
+variable "nutanix_username" {
+  description = "The username used to authenticate with Prism Central."
+  type        = string
+}
+
+variable "nutanix_password" {
+  description = "The password used to authenticate with Prism Central."
+  type        = string
+}
+
+variable "nutanix_endpoint" {
+  description = "The endpoint URL for Prism Central."
+  type        = string
+}
+
+variable "nutanix_port" {
+  description = "The port to use when connecting to Prism Central."
+  type        = number
+  default     = 9440
+}
+
+variable "nutanix_insecure" {
+  description = "Set to true to disable SSL certificate validation, false by default."
+  type        = bool
+  default     = false
+}
+
+variable "nutanix_cluster" {
+  description = "The name of the Nutanix cluster to deploy to."
+  type        = string
+}
+
+variable "nutanix_subnet" {
+  description = "The name of the subnet to deploy VMs to."
+  type        = string
+}
+
+variable "name" {
+  description = "Name to use for VMs and other resources."
+  type        = string
+  default     = "rke2-server"
+}
+
+variable "unique_suffix" {
+  description = "Whether to add a unique suffix to the name."
+  type        = bool
+  default     = true
+}
+
+variable "server_count" {
+  description = "The number of server VMs to deploy."
+  type        = number
+  default     = 3
+}
+
+variable "agent_count" {
+  description = "The number of agent VMs to deploy."
+  type        = number
+  default     = 6
+}
+
+
+variable "server_memory" {
+  description = "The amount of memory for each server VM in MiB."
+  type        = number
+}
+
+variable "server_cpu" {
+  description = "The number of CPUs for each server VM."
+  type        = number
+}
```

```suggestion
variable "server_sockets" {
  description = "The number of CPUs for each server VM."
  type        = number
  default    = 1
}
```

Change verbage to match Nutanix verbage. Makes it more clear what setting this maps to. Also default to 1 since most of the time you probably aren't creating a VM with multiple sockets

---

Adding comment here from Slack - not sure on what is best here. The Nutanix/Prism Central console actually shows these as CPUs (and the other field as Cores) so I modeled this var after that. I'd lean towards that being more straightforward wording but I'm okay with this change if you disagree. It was annoying trying to understand what this tf provider config mapped to when making a VM just via the UI so I tried to keep the vars 1:1 with what you'd see in the UI vs what you'd see just using the TF provider yourself.

---

Also not sure on the default value...based on reading from [Nutanix docs](https://portal.nutanix.com/page/documents/solutions/details?targetId=BP-2029-AHV:cpu-configuration.html) I was seeing this:

> AHV lets you configure vCPUs (equivalent to CPU sockets) and cores per vCPU (equivalent to CPU cores) for each VM. Nutanix recommends first increasing the number of vCPUs when a VM needs more than 1 CPU, rather than increasing the cores per vCPU. For example, if a guest VM requires 4 CPU, Nutanix recommends using 4 vCPU with one core per vCPU.

That would lead me to thinking that its better to have a default core count of 1 and no default for socket/cpu?